lib/msf/core/db_manager/import/metasploit_framework/xml.rb

```diff
@@ -229,7 +229,7 @@ def import_msf_web_vuln_element(element, options={}, &notifier)
   # TODO: loot, tasks, and reports
   def import_msf_xml(args={}, &block)
     data = args[:data]
-    wspace = args[:wspace] || workspace || Msf::Util::DBManager.process_opts_workspace(args, framework).name
+    wspace = args[:wspace] || args[:workspace] || Msf::Util::DBManager.process_opts_workspace(args, framework).name
```

Comment:
The recursion was due to what `workspace` ended up calling?

Reply:
There is no reason any of the DbManager `import_*` methods should be using a pattern similar to `wspace = args[:wspace] || args[:workspace] || Msf::Util::DBManager.process_opts_workspace(args, framework).name`. It is redundant as `Msf::Util::DBManager.process_opts_workspace` calls the method `delete_opts_workspace` that pulls from `: workspace` first and then `:wspace`.

Reply:
If you need the workspace in DbManager methods simply use the helper as follows:
```
wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework)
```

Reply:
`:workspace` argument is set here.
https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/db_manager/import.rb#L97

Recursion occurring here when using the service.
https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/db_manager/workspace.rb#L30